### PR TITLE
add W&I + LOCNESS dataset (BEA-2019 workshop shared task on GEC) [PROPER]

### PR DESCRIPTION
**Pull request by aseifert**
- **Name:** W&I + LOCNESS dataset (from the BEA-2019 workshop shared task on GEC)
- **Description:** https://www.cl.cam.ac.uk/research/nl/bea2019st/#data
- **Paper:** https://www.aclweb.org/anthology/W19-4406/
- **Motivation:** This is a recent dataset (actually two in one) for grammatical error correction and is used for benchmarking in this field of NLP.

### Checkbox

- [x] Create the dataset script `/datasets/my_dataset/my_dataset.py` using the template
- [x] Fill the `_DESCRIPTION` and `_CITATION` variables
- [x] Implement `_infos()`, `_split_generators()` and `_generate_examples()`
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is filled with the different configurations of the dataset and that the `BUILDER_CONFIG_CLASS` is specified if there is a custom config class.
- [x] Generate the metadata file `dataset_infos.json` for all configurations
- [x] Generate the dummy data `dummy_data.zip` files to have the dataset script tested and that they don't weigh too much (<50KB)
- [x] Add the dataset card `README.md` using the template : fill the tags and the various paragraphs
- [x] Both tests for the real data and the dummy data pass.

### CODE REVIEW

**Comment by tanmoyio** [comment]
linter your code with flake8 and also run the commands present in Makefile for proper formatting

File: datasets/wi_locness/wi_locness.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        data_dir = Path(dl_manager.download_and_extract(_URL)) / "wi+locness/json"
```
Comment (lhoestq) [inline datasets/wi_locness/wi_locness.py:164]:
```suggestion
        data_dir = Path(dl_manager.download_and_extract(_URL)) / "wi+locness" / "json"
```

File: datasets/wi_locness/wi_locness.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+        if self.config.name == "wi":
+            levels = ["A", "B", "C"]
+        elif self.config.name == "locness":
+            levels = ["N"]
+        else:
+            assert False
+
+        for level in levels:
+            with open(filepath / f"{level}.{split}.json", "r") as fp:
```
Comment (lhoestq) [inline datasets/wi_locness/wi_locness.py:204]:
```suggestion
            with open(filepath / f"{level}.{split}.json", "r", encoding="utf-8") as fp:
```

**Review by lhoestq — COMMENTED** [review]
Really cool thanks !

**Review by lhoestq — APPROVED** [review]
LGTM thanks :)

**Pull request merged**